com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CollectSystemParametersStep.java

```diff
@@ -68,12 +70,14 @@ protected String getStepErrorMessage(ProcessContext context) {
         return Messages.ERROR_COLLECTING_SYSTEM_PARAMETERS;
     }
 
-    private String getDefaultDomain(CloudControllerClient client) {
-        CloudDomain defaultDomain = client.getDefaultDomain();
-        if (defaultDomain != null) {
-            return defaultDomain.getName();
+    private String getDefaultDomain(CloudControllerClient client, ProcessContext context) {
+        try {
+            return client.getDefaultDomain()
+                         .getName();
+        } catch (CloudOperationException exc) {
```

Comment:
What if CloudOPerationException is thrown for other reaseon?
I think, we will consider that default domain is missing where there was a temporary issue with controller, for example.

Reply:
For that to happen, the exception must be thrown 4 times. This is our retryable client, so if a timeout, or a network issue occurs, it will retry automatically. But for a missing default domain, it will always fail with this exception.

Reply:
I cannot fully agree with that because client is defined as CloudControllerClient, not as some special RetryableClient or anything else that have some logic.
Also CloudControllerClient does not declare throwing of such exception.
So if the retry failed all 4 times, what will happen. I guess we will continue silently with the DEFAULT_DOMAIN_PLACEHOLDER, right?